util/shared/modules.mjs

```diff
@@ -303,6 +336,108 @@ const createSetterFromProperty = property => {
     return setter
 }
 
+const createReadyFromProvider = provider => {
+
+    if (provider.name.startsWith('onRequest')) {
+        const ready = JSON.parse(JSON.stringify(provider))
+        ready.name = ready.name.charAt(9).toLowerCase() + ready.name.substr(10) + 'Ready'
+        ready.summary = `Internal API for ${provider.name.substr(9)} Provider to notify when ready.`
+        const old_tags = ready.tags
+        ready.tags = [
+            {
+                'name': 'rpc-only',
+                'x-handshake-for': provider.name
+            }
+        ]
+
+        ready.params = []
+        ready.result = {
+            name: 'result',
+            schema: {
+                type: "null"
+            }
+        }
+
+        ready.examples = [
+            {
+                name: "Example",
+                params: [],
+                result: {
+                    name: "result",
+                    value: null
+                }
+            }
+        ]
+
+        return ready
+    }
+    else {
+        throw "Methods with the `x-provider` tag extension MUST start with 'onRequest'."
+    }
+}
+
+const createResponseFromProvider = (provider, json) => {
+
+    if (provider.name.startsWith('onRequest')) {
```

Comment:
minor syntax improvement:
```
if (provider.name.startsWith('onRequest')) {
   throw "Methods with the `x-provider` tag extension MUST start with 'onRequest'."
}
// function code
```
That way the entire function doesn't need to be indented

Reply:
Same comment for `createReadyFromProvider`

Reply:
you meant to put a `!` at the start of the `if`, right? Will do.